drv/cosmo-seq-server/src/main.rs

```diff
@@ -686,7 +719,27 @@ impl ServerImpl {
         // we probably(?) won't see multiple of these set at a time but
         // it's important to account for that case;
 
-        let mut action = InternalAction::None;
+        let mut action = InternalAction::Unexpected;
+
+        if ifr.pwr_cont1_to_fpga1_alert || ifr.pwr_cont2_to_fpga1_alert {
+            // We got a PMBus alert from one of the Vcore regulators.
+            let now = sys_get_timer().now;
+            ringbuf_entry!(Trace::PmbusAlert { now });
+            let which_rails = vcore::Rails {
+                vddcr_cpu0: ifr.pwr_cont1_to_fpga1_alert,
+                vddcr_cpu1: ifr.pwr_cont2_to_fpga1_alert,
+            };
+            self.vcore.handle_pmbus_alert(which_rails, now);
+
+            // We need not instruct the sequencer to reset. PMBus alerts from
+            // the RAA229620As are divided into two categories, "warnings" and
+            // "faults", where "warnings" just pull PMALERT_L and set status
+            // bits, and "faults" also cause the VRM to deassert POWER_GOOD. If
+            // POWER_GOOD is deasserted, the sequencer FPGA will notice that and
+            // generate a subsequent IRQ, which is handled separately. So, all
+            // we need to do here is proceed and handle any other interrupts.
+            action = InternalAction::None;
+        }
```

Comment:
I don't see a clear of `ifr` anywhere, si that handled elsewhere?

Reply:
I left a comment regarding this, but at some point while moving things around, it ended up in the `vcore.rs` file; it probably should have stayed in this file:
https://github.com/oxidecomputer/hubris/blob/238ff27f36aaed5755a78847de562d5678b4efbe/drv/cosmo-seq-server/src/vcore.rs#L224-L240

Essentially, these bits behave somewhat differently from the sequencer-generated IRQs; they're set if `PMALERT_L` is asserted, and are just passed through blindly by the FPGA; the IRQ is cleared by clearing the fault in the regulator so that it deasserts PMALERT_L, rather than by telling the FPGA to clear it.